### ISSUE TITLE
PublishCodeCoverageResult is broken for JaCoCo after Autogenerate HTML reports for Code Coverage #9704

### ISSUE THREAD

**Issue by in4margaret** [microsoft/azure-pipelines-tasks#10212]
Publishing of JaCoCo coverage results is broken after #9704. 
If you have a look at https://github.com/Microsoft/azure-pipelines-tasks/pull/9704/files#diff-3afbe858d80311a2afe7be02b22dedd4R87 it uses 'tempFolder' variable as report directory but the only place it's value is defined is inside of this 'if' https://github.com/Microsoft/azure-pipelines-tasks/pull/9704/files#diff-3afbe858d80311a2afe7be02b22dedd4R44. Note that autogenerateHtmlReport is 
```ts
let autogenerateHtmlReport: boolean = codeCoverageTool === 'Cobertura';
```
so effectively this PR broke publishing of JaCoCo reports. 
It's also strange that changing functionality in not backward compatible way(and removing the option to provide your own report directory and forcing to have dotnet installed is not in any way backward compatible change) resulted in change of a minor part of a semver version.

**Comment by karanjitsingh** [comment]
@in4margaret, We have a fix ready for this, for jacoco to work with autogenerated html reports, you have to specify path to source files as well. We've added this field in the publish code coverage task

![image](https://user-images.githubusercontent.com/4632805/59016395-c1b9e680-885e-11e9-917d-90b23ac90565.png)

**Comment by karanjitsingh** [comment]
Closing this thread, if you're still facing problems, please create a new issue referencing this thread.
